.github/workflows/e2e-test-cron.yml

```diff
@@ -0,0 +1,89 @@
+name: E2E Tests Nightly
+
+on:
+  schedule:
+    # Runs "At 01:00 every night except weekends"
+    - cron: '0 1 * * 1-5'
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  build-compose-apks:
+    name: Build
+    uses: ./.github/workflows/e2e-build.yml
+    with:
+      app: compose
+
+  run-compose-tests-nightly:
+    name: Test compose
+    runs-on: ubuntu-24.04
+    needs: build-compose-apks
+    strategy:
+      matrix:
+        include:
+          - android_api_level: 35
+          - android_api_level: 34
+          - android_api_level: 33
+          - android_api_level: 32
+          - android_api_level: 31
+          - android_api_level: 29
+          - android_api_level: 28
+      fail-fast: false
+    env:
+      ANDROID_API_LEVEL: ${{ matrix.android_api_level }}
+    steps:
+      - name: Connect Bot
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.BOT_SSH_PRIVATE_KEY }}
+      - uses: actions/checkout@v4.2.2
+      - uses: actions/download-artifact@v4.1.8
+        continue-on-error: true
+        with:
+          name: apks
+      - uses: ./.github/actions/java-cache
+      - uses: ./.github/actions/enable-kvm
+      - uses: ./.github/actions/ruby-cache
+      - uses: ./.github/actions/gradle-cache
+        with:
+          key-prefix: gradle-test
```

Comment:
Do we need to get the gradle cache even we are not running any gradle task?
If I understand it correctly, we are compiling the app on the previous job with the `e2e-build` action

Reply:
Fair point, removed gradle cache for testing jobs.